### PR TITLE
Add JSONLoader to allow customization of loaded JSON config

### PR DESCRIPTION
**Pull request by dissoupov**

### CODE REVIEW

File: cmd/configen/config.go.template [hunk 1]
```diff
@@ -25,6 +25,12 @@ import (
 	"time"
 )
 
+// LoadJSONFunc defines a function type to load JSON configuration file
+type LoadJSONFunc func (filename string, v interface{}) error
+
+// JSONLoader allows to speciy a custom loader
```
Comment (haykbaluyan) [inline cmd/configen/config.go.template:31]: specify -typo

File: cmd/configen/config.go.template [hunk 2]
```diff
@@ -25,6 +25,12 @@ import (
 	"time"
 )
 
+// LoadJSONFunc defines a function type to load JSON configuration file
+type LoadJSONFunc func (filename string, v interface{}) error
```
Comment (haykbaluyan) [inline cmd/configen/config.go.template:29]: just curious why this needs to be exported? If not exported, can't you still assigned values to the JSONLoader variable?

Comment (dissoupov) [reply cmd/configen/config.go.template:29]: If var is exported, it's type should be also public

File: cmd/configen/config_test.go.template [hunk 3]
```diff
@@ -180,3 +181,46 @@ func Test_LoadInvalidJson(t *testing.T) {
...
+        return err
+    }
+
+    val := strings.ReplaceAll(string(bytes),"${ENV}", "ENV_VALUE")
+    return json.NewDecoder(strings.NewReader(val)).Decode(v)
+}
+
+func Test_LoadCustomJSON(t *testing.T) {
+{{ $t := (index .Structs "Configuration").GoType }}
+  c := Configurations{
```
Comment (pogodin) [inline cmd/configen/config_test.go.template:197]: Seems like there is no space after `{`. Isn't this code formatted?

File: cmd/configen/config_test.go.template [hunk 4]
```diff
@@ -180,3 +181,46 @@ func Test_LoadInvalidJson(t *testing.T) {
...
+    Defaults: {{index $t.ExampleValues 1}},
+    Hosts : map[string]string{ "bob" : "${ENV}"},
+    Overrides : map[string]Configuration{
```
Comment (pogodin) [inline cmd/configen/config_test.go.template:200]: Same thing here: no space before `{`

File: cmd/configen/config_test.go.template (hunk 3 above)
Comment (pogodin) [reply cmd/configen/config_test.go.template:197]: Never mind. Seems like this IS a GoLang formatting

File: cmd/configen/config_test.go.template (hunk 4 above)
Comment (dissoupov) [reply cmd/configen/config_test.go.template:200]: this is template. if should be run under `go fmt` on the final output.

**Pull request merged**